--- a/charon/src/transform/resugar/reconstruct_panic_calls.rs
+++ b/charon/src/transform/resugar/reconstruct_panic_calls.rs
@@ -12,6 +12,7 @@
 impl Transform {
     pub fn new(ctx: &TransformCtx) -> CowBox<dyn UllbcPass> {
         let explicit_panic = Name::from_path(names::EXPLICIT_PANIC_NAME);
+        let panic_nounwind_fmt = Name::from_path(&["core", "panicking", "panic_nounwind_fmt"]);
         let assert_failed = Name::from_path(&["core", "panicking", "assert_failed"]);
         let panic_fns = ctx
             .translated
@@ -30,7 +31,8 @@
                 );
                 (is_panic_lang_item
                     || decl.item_meta.name == explicit_panic
-                    || decl.item_meta.name == assert_failed)
+                    || decl.item_meta.name == assert_failed
+                    || decl.item_meta.name == panic_nounwind_fmt)
                     .then(|| (id, decl.item_meta.name.clone()))
             })
             .collect();
